src/main.rs

```diff
@@ -89,20 +93,33 @@ impl Service for Contributors {
                     f.read_to_string(&mut source).unwrap();
 
                     use contributors::schema::commits::dsl::*;
+                    use contributors::models::Commit;
 
                     let connection = contributors::establish_connection();
 
-                    let mut names: Vec<String> = commits.select(author_name).distinct().load(&connection).unwrap();
+                    // It's possible to do this in Postgres with
+                    // SELECT author_name, COUNT(author_name) as commit_count FROM commits GROUP BY author_name ORDER BY commit_count DESC;
+                    // but it doesn't look like Diesel supports that
```

Comment:
Okay! It's not pretty, but it works:

```rust
let mut scores: Vec<(String, i64)> =
    commits
    .select((author_name, sql::<BigInt>("COUNT(author_name) AS author_count")))
    .group_by(author_name)
    .order(sql::<BigInt>("author_count").desc())
    .load(&connection)
    .unwrap();
```

I think there is a better way to get the "count as …", but `.aliased` let me down and I didn't want to debug this right now. `group_by` was added a year ago, but is hidden in the docs to [show](https://github.com/diesel-rs/diesel/commit/6df20dee3191c71b9fff80097eb0058f302b7e3d) it's experimental.